### PR TITLE
[SYCL][COMPAT] Added helpers for occupancy calculation in Intel GPUs

### PR DESCRIPTION
**Pull request by Alcpz**
Adds helpers to calculate the occupancy of the GPU

Testing currently only addresses API.

### CODE REVIEW

File: sycl/doc/syclcompat/README.md [hunk 1]
```diff
@@ -1146,6 +1152,18 @@ public:
...
 };
 
+inline int calculate_max_active_wg_per_xecore(int *num_wg, int wg_size,
+                                              int slm_size = 0,
+                                              int sg_size = 32,
+                                              bool used_barrier = false,
+                                              bool used_large_grf = false);
+
+inline int calculate_max_potential_wg(int *num_wg, int *wg_size,
+                                      int max_ws_size_for_device_code,
```
Comment (AidanBeltonS) [inline sycl/doc/syclcompat/README.md:1162]:
```suggestion
                                      int max_wg_size_for_device_code,
```
Possible typo?

File: sycl/include/syclcompat/util.hpp [hunk 2]
```diff
@@ -404,5 +404,130 @@ class logical_group {
...
+}
+
+/// This function is used for occupancy calculation, it computes the work-group
+/// number and the work-group size which achieves the maximum occupancy of the
+/// device potentially. Ref to
+/// https://github.com/oneapi-src/oneAPI-samples/tree/master/Tools/GPU-Occupancy-Calculator
+/// \param [out] num_wg Work-group number.
+/// \param [out] wg_size Work-group size.
+/// \param [in] max_ws_size_for_device_code The maximum working work-group size
+/// for current device code logic. Zero means no limitation.
```
Comment (AidanBeltonS) [inline sycl/include/syclcompat/util.hpp:501]: Possible typo, just apply this comment to the function below

**Review by AidanBeltonS — APPROVED** [review]
Fine outside of NIT

**Comment by Alcpz** [comment]
@intel/llvm-gatekeepers PR is ready for merging. Thanks!

**Pull request merged**